src/backend/distributed/connection/remote_commands.c

```diff
@@ -459,6 +433,33 @@ ExecuteOptionalRemoteCommand(MultiConnection *connection, const char *command,
 }
 
 
+/*
+ * ExecuteCriticalRemoteCommand executes a remote command that is critical
+ * to the transaction. If the command fails then the transaction aborts.
+ */
+void
+ExecuteCriticalRemoteCommand(MultiConnection *connection, const char *command,
+							 PGresult **result)
```

Comment:
I think modifying this function is no longer necessary for this PR. Maybe leave it to another PR (or not since it'll be dead code).

Reply:
Reverted. Personally think the implementation being shared between Critical & Optional is correct, but the third parameter would be omitted in a PR